### PR TITLE
feat: allow single dollar delimiters

### PR DESCRIPTION
**Pull request by arzola**
Address https://github.com/pressbooks/pb-mathjax/issues/124

Add user-facing option to support single dollar sign delimiters (`$...) for inline LaTeX math, with careful handling to avoid false positives (such as currency). Update backend logic and MathJax integration, and add tests.

**Single Dollar Delimiter Support**

* Add a new `use_single_dollar` option for MathJax, allowing users to enable or disable `$... as inline math delimiters. This option is exposed in the admin interface with clear documentation and is off by default. [[1]](diffhunk://#diff-75471e261b56521dd46690fedb88783fbf2ce63be3148866d236767dfe25fde2L33-R37) [[2]](diffhunk://#diff-75471e261b56521dd46690fedb88783fbf2ce63be3148866d236767dfe25fde2R185) [[3]](diffhunk://#diff-75471e261b56521dd46690fedb88783fbf2ce63be3148866d236767dfe25fde2R207-R211) [[4]](diffhunk://#diff-183101177c03e8c9991d2f31dada2f1add4f4a66a9f461647b5bad5c4fe1cbb4R49-R64)
* When enabled, server-side processing (not MathJax config) converts `$...` to `\(...\)` using a regex that avoids matching currency or escaped dollar signs, ensuring only intended math expressions are affected.
* The `sectionHasMath` method now detects inline math using single dollar delimiters only if the option is enabled, with logic to avoid currency and similar patterns.
* The MathJax configuration is unchanged for single dollar delimiters—processing is handled server-side to avoid MathJax's greedy matching issues. [[1]](diffhunk://#diff-75471e261b56521dd46690fedb88783fbf2ce63be3148866d236767dfe25fde2L278-R294) [[2]](diffhunk://#diff-75471e261b56521dd46690fedb88783fbf2ce63be3148866d236767dfe25fde2L308-R341)
* Add a filter hook (`pb_mathjax_use_single_dollar`) for programmatic override of this setting.

**Admin Interface Improvements**

* The MathJax settings page now includes a checkbox to enable/disable single dollar delimiter support, with user guidance and examples. The feature is also documented in the "Allowed Syntax" section, which conditionally displays the `$... syntax when enabled. [[1]](diffhunk://#diff-183101177c03e8c9991d2f31dada2f1add4f4a66a9f461647b5bad5c4fe1cbb4L20-R29) [[2]](diffhunk://#diff-183101177c03e8c9991d2f31dada2f1add4f4a66a9f461647b5bad5c4fe1cbb4R49-R64)

Pasted_Image_2026-04-02__9_04 PM

**Testing and Quality Assurance**

* Add tests to verify default behavior, option toggling, correct detection of math and currency, export pipeline integration, and edge cases (e.g., escaped dollars, display math, mixed content). [[1]](diffhunk://#diff-75c656a671caff55ad7a4836ccb66dd2b86b860241349385d5c8978feeb65006R56-R90) [[2]](diffhunk://#diff-75c656a671caff55ad7a4836ccb66dd2b86b860241349385d5c8978feeb65006R228-R376) [[3]](diffhunk://#diff-75c656a671caff55ad7a4836ccb66dd2b86b860241349385d5c8978feeb65006R388-R443)

### CODE REVIEW

File: templates/admin/mathjax.blade.php [hunk 1]
```diff
@@ -10,36 +10,58 @@
...
             </tr>
+            <tr>
+                <th scope="row">
+                    <label for="mathjax-use-single-dollar">{{ __( 'Single dollar sign delimiter', 'pressbooks' ) }}</label>
+                </th>
+                <td>
+                    <input type="checkbox" name="use_single_dollar" id="mathjax-use-single-dollar"
+                           value="1" @if ( $use_single_dollar ) checked @endif />
+                    <p>
+                        {!! sprintf( __( 'When enabled, %s will be treated as inline LaTeX math.', 'pressbooks' ), '<code>$e^{i \pi} + 1 = 0$</code>' ) !!}
```
Comment (SteelWagstaff) [inline templates/admin/mathjax.blade.php]:
```suggestion
                        {!! sprintf( __( 'When enabled, %1$s will be treated as inline LaTeX math. Use %2$s to display a literal dollar sign.', 'pressbooks'	), '<code>$e^{i \\pi} + 1 = 0$</code>', '<code>\\$</code>' ) !!}
                    </p>
                    <p>
                        {{ __( 'Note: The opening $ must not be followed by a space and the closing $ must not be preceded by a space.', 'pressbooks' ) }}
```

**Review by fdalcin — APPROVED** [review]
LGTM
